crates/proof-of-sql/src/base/bit/bit_distribution.rs

```diff
@@ -73,92 +90,24 @@ impl BitDistribution {
-        // handle the case of everything zero
-        if self.num_varying_bits() == 0 && self.constant_part() == [0; 4] {
-            return true;
-        }
-
         // signed 128 bit numbers range from
         //      -2^127 to 2^127-1
         // the maximum absolute value of the sum of two signed 128-integers is
         // then
         //       2 * (2^127) = 2^128
-        self.most_significant_abs_bit() <= 128
-    }
-
-    /// If `{b_i}` represents the non-varying 1-bits of the absolute values, return the value
-    ///    `sum_i b_i 2 ^ i`
-    pub fn constant_part(&self) -> [u64; 4] {
-        let mut val = [0; 4];
-        self.for_each_abs_constant_bit(|i: usize, bit: usize| {
-            val[i] |= 1u64 << bit;
-        });
-        val
-    }
-
-    /// Iterate over each constant 1-bit for the absolute values
-    pub fn for_each_abs_constant_bit<F>(&self, mut f: F)
-    where
-        F: FnMut(usize, usize),
-    {
-        for i in 0..4 {
-            let bitset = if i == 3 {
-                !(self.vary_mask[i] | (1 << 63))
-            } else {
-                !self.vary_mask[i]
-            };
-            let bitset = bitset & self.or_all[i];
-            for pos in BitIter::from(bitset) {
-                f(i, pos);
-            }
-        }
-    }
-
-    /// Iterate over each varying bit for the absolute values
-    pub fn for_each_abs_varying_bit<F>(&self, mut f: F)
-    where
-        F: FnMut(usize, usize),
-    {
-        for i in 0..4 {
-            let bitset = if i == 3 {
-                self.vary_mask[i] & !(1 << 63)
-            } else {
-                self.vary_mask[i]
-            };
-            for pos in BitIter::from(bitset) {
-                f(i, pos);
-            }
-        }
-    }
-
-    /// Iterate over each varying bit for the absolute values and the sign bit
-    /// if it varies
-    pub fn for_each_varying_bit<F>(&self, mut f: F)
-    where
-        F: FnMut(usize, usize),
-    {
-        for i in 0..4 {
-            let bitset = self.vary_mask[i];
-            for pos in BitIter::from(bitset) {
-                f(i, pos);
-            }
-        }
+        (self.leading_bit_inverse_mask() >> 128) == (U256::MAX >> 129)
     }
 
-    /// Return the position of the most significant bit of the absolute values
+    /// Iterate over each varying bit
+    ///
     /// # Panics
-    /// Panics if no bits are set to 1 in the bit representation of `or_all`.
-    pub fn most_significant_abs_bit(&self) -> usize {
-        let mask = self.or_all[3] & !(1 << 63);
-        if mask != 0 {
-            return 64 - (mask.leading_zeros() as usize) - 1 + 3 * 64;
-        }
-        for i in (0..3).rev() {
-            let mask = self.or_all[i];
-            if mask != 0 {
-                return 64 - (mask.leading_zeros() as usize) - 1 + 64 * i;
-            }
-        }
-        panic!("no bits are set");
+    ///
+    /// The panic shouldn't be mathematically possible
```

Comment:
NIT: If this is true, then use `#[allow(clippy::missing_panic_doc)]` to ignore the lint.
You only need to document panics if you can actually panic.